src/scheduler/resv_info.c

```diff
@@ -1589,6 +1594,9 @@ confirm_reservation(status *policy, int pbs_sd, resource_resv *unconf_resv, serv
 		if (!(simrc & TIMED_ERROR) && resv_start_time >= 0) {
 			clear_schd_error(err);
 			if ((ns = is_ok_to_run(nsinfo->policy, nsinfo, NULL, nresv, NO_ALLPART, err)) != NULL) {
+				if (vnodes_down == 1)
```

Comment:
But we don't sort in the cases when check_vnodes_unavailable returns -1 or -2. 
I think maybe I should add a check in check_vnodes_unavailable() to sort only when we see a vnode unavailable and avoid sorting when all vnodes are up.
Since we don't sort in all cases, I thought we don't have to re-sort to bring nspec back in the original order.